### PR TITLE
Refactor config

### PR DESCRIPTION
**Pull request by cappyzawa**
## Context

There was a lot of duplicate code.
If a single element is added to the config (entry), the change will be very costly.

## Objective

Simplify the config to make it easier to add features.

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -163,31 +169,12 @@ func (c *Config) Save() error {
...
-		e.UUID = value
-	default:
+	var m map[string]interface{}
+	mapstructure.Decode(e, &m)
+	if _, ok := m[key]; !ok {
 		return fmt.Errorf("invalid key %s", key)
 	}
-
+	m[key] = value
+	mapstructure.Decode(m, &e)
```
Comment (tanoda) [inline config/config.go:178]: This code should work on update value (except empty value handling), although I suggest add some comments to explain how works.
1. encode current configuration to empty map
2. check map key found (error handling for unknown key) and set value
3. update current configuration by map

File: config/config.go [hunk 2]
```diff
@@ -163,31 +169,12 @@ func (c *Config) Save() error {
...
-		}
-		e.Launcher.Image = value
-	case "uuid":
-		if value == "" {
-			value = "-"
-		}
-		e.UUID = value
-	default:
+	var m map[string]interface{}
+	mapstructure.Decode(e, &m)
```
Comment (yuichi10) [inline config/config.go:173]: Do you need to check error here?

Comment (cappyzawa) [reply config/config.go:177]: OK.
I will add error handling.

However, I don't know in what case the error occurs.

File: cmd/config/create.go [hunk 3]
```diff
@@ -26,17 +26,18 @@ The new config has only launcher-version and launcher-image.`,
...
 
-			config, err := configNew(path)
+			c, err := configNew(path)
 			if err != nil {
 				return err
 			}
 
-			err = config.AddEntry(name)
+			defaulEntry := config.DefaultEntry()
+			err = c.AddEntry(name, defaulEntry)
```
Comment (tkyi) [inline cmd/config/create.go:35]:
```suggestion
			defaultEntry := config.DefaultEntry()
			err = c.AddEntry(name, defaultEntry)
```

**Review by yuichi10 — APPROVED** [review]
LGTM

**Review by tkyi — APPROVED** [review]

**Pull request merged**